pkg/storage/stores/shipper/downloads/table.go

```diff
@@ -413,3 +418,73 @@ func getDBNameFromObjectKey(objectKey string) (string, error) {
 	}
 	return ss[1], nil
 }
+
+// doParallelDownload downloads objects(dbs) parallelly. It is upto the caller to open the dbs after the download finishes successfully.
+func (t *Table) doParallelDownload(ctx context.Context, objects []chunk.StorageObject) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	folderPath, err := t.folderPathForTable(true)
+	if err != nil {
+		return err
+	}
+
+	queue := make(chan chunk.StorageObject)
+	n := util.Min(len(objects), downloadParallelism)
+	incomingErrors := make(chan error, n)
```

Comment:
I'm wondering if this should not be a buffered channel, honestly I'm not sure it will really make much difference because of how fast the loop should be that reads from it.

By having it buffered one of the worker threads could receive an error and then immediately start downloading the next item from the queue.

If we make it unbuffered then the check for success becomes synchronous over all the worker threads and any errors encountered would result in immediate cancel before workers started another download.

I think it makes more sense to me to remove the buffering on this channel.